### DISCUSSION TITLE
Separate permission for security-events that allows publishing without reading all other events

### DISCUSSION THREAD

**Discussion by sethmlarson** [community/community#29710] [Apps, API and Webhooks]
Currently `security-events: write` is the only way for a GitHub Action to create security events. However having `write` implies the GitHub Action has the ability to read security events as well which has the potential to disclose vulnerabilties to a GitHub Action that's taken over.

My gut feeling is that almost all GitHub Actions that require `security-events: write` don't actually need `security-events: read` because they only want to publish a security event, not view any already published security events.

I discovered this issue when reporting https://github.com/ossf/scorecard/issues/2152 which is where OpenSSF Scorecard will dock points from a repository using *it's own GitHub Action* because having `security-events: write` is insecure, but it requires the write permission in order to publish it's own findings!

**Comment by jsoref** [comment]
I'd really love a `security-events-create: write` which would allow an entity with a specific token to create and manipulate only the events it created...

I'm personally just as concerned as everyone else is by the wide-ranging permissions granted to carriers of the `security-events: write` permission and that wide-ranging grant is causing potential users of my action to push back against enabling the feature.

**Comment by Peter-Darton-i2** [comment]
I've recently encountered this from the usecase of wanting my CI builds to upload CodeQL results.
I do not wish to give my CI builds carte-blanche to rewrite my security events; that seems somewhat foolish and violates our security principles.
I only want them to be able to upload CodeQL results (for their own build, for their own commit) but the existing implementation does not permit this.

i.e. in order to improve code security, one needs to give builds "the keys to the castle" and just hope that nobody commits code that misuses the access the CI builds are given.

This does _not_ follow security best-practise of giving things the least privilege/access they require.
